SimPEG/electromagnetics/frequency_domain/survey.py

```diff
@@ -97,3 +106,34 @@ def get_sources_by_frequency(self, frequency):
             frequency in self._frequency_dict
         ), "The requested frequency is not in this survey."
         return self._frequency_dict[frequency]
+
+    @property
+    def source_location_by_sounding_dict(self):
```

Comment:
Type the return property, so no need to put it in the name.
```suggestion
    def source_location_by_sounding(self) -> dict:
```